### DISCUSSION TITLE
Interactive Plot

### DISCUSSION THREAD

**Discussion by wktanghk** [scikit-rf/scikit-rf#813] [Q&A] [answered]
Hi, I would like to seek recommendations on how to generate interactive plots. For example, given a 6-port network, I would like to interactively choose to plot dB(S21), dB(S31),etc, perhaps through a drop-down menu. Ideally, I intend to run this code as a tkinter GUI, not through Jupyter notebook. Currently, with the built-in Matplotlib, I'm not able to have such interaction. Once the plot is created, I am not able to allow user to add or remove traces to an existing plot.

I think this may not be a scikit-rf specific question. So if this is not the right forum to post, I would gladly remove this. Just like to see if anyone could help point to some directions. Thanks.

**Comment by jhillairet** [comment]
Have a look to the `qt_apps`  directory or scikit-rf for examples of QT applications, or also to https://github.com/scikit-rf/dash-apps for example of DASH applications.

**Comment by JAnderson419** [comment, marked as answer]
I don't know of any Tk implementations of skrf specifically, but you might be able to find inspiration from the qtapps or from the [dash app](https://github.com/scikit-rf/dash-apps) I developed, which should be able to handle your 6-port files. Dash is a good enterprise solution, but is probably overkill for a GUI for a small script. If you are just trying to write a GUI for a small script and are not set on Tk, I have recently used [streamlit](https://streamlit.io) and have enjoyed it. They provide a good way to quickly convert smaller scripts to a GUI without having to write callback functions. Like Dash, this runs in the browser using plotly as the backend. Plotly does have a matplotlib converter built in, but it sometimes has trouble depending on how complex your plots are. I have not tried smith charts in plotly since they added support a year or two ago, but I had rectangular plots working well in my Dash app.

Hope that helps.

**Reply by jhillairet** [reply]
how, what a synchronism ! ;)

**Reply by wktanghk** [reply]
Thank you both so much @JAnderson419 & @jhillairet. These are great pointers!
